docs-src/httproute.md

```diff
@@ -0,0 +1,303 @@
+metadata:
+  name: httproute-tls
+spec:
+  gatewayClassName: example-gc
+  listeners:
+  - protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      ...
+      routeOverride:
+        certificate: Allow
+---
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: http-app-1
+spec:
+  hostnames:
+  - "foo.example.com"
+  tls:
+    certificateRef:
+      kind: Secret
+      group: core
+      name: foo-example-com-cert
+```
+
+`CertificateRef` refers to a Kubernetes object that contains a TLS certificate and private key. This certificate MUST be
+used for TLS handshakes for the domain this `tls` is associated with. If an entry in this list omits or specifies the
+empty string for both the `group` and `kind`, the resource defaults to “secrets”.
+
+**Notes:**
+
+- HTTPRoute selection takes place after the TLS Handshake (ClientHello). Due to this, a TLS certificate
+  in an HTTPRoute will take precedence even if the request has the potential to match multiple HTTPRoutes (in case
+  multiple HTTPRoutes share the same hostname).
+- Collisions can happen if multiple HTTPRoutes define a TLS certificate for the same hostname. In such case, the
+  certificate in the oldest HTTPRoute is selected.
+
+### Rules
+
+Rules define semantics for matching an HTTP request based on conditions, optionally executing additional processing
+steps, and optionally forwarding the request to an API object.
+
+#### Matches
+
+Matches define conditions used for matching an HTTP request. Each match is independent, i.e. this rule will be matched if
+any single match is satisfied.
+
+Take the following matches configuration as an example:
+```yaml
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+...
+matches:
+  - path:
+      value: "/foo"
+    headers:
+      values:
+        version: "2"
+  - path:
+      value: "/v2/foo"
+```
+
+For a request to match against this rule, it must satisfy EITHER of the following conditions:
+
+ - A path prefixed with /foo **AND** contains the header "version: 2"
+ - A path prefix of /v2/foo
+
+If no matches are specified, the default is a prefix path match on “/”, which has the effect of matching every HTTP
+request.
+
+#### Filters (optional)
+
+Filters define processing steps that must be completed during the request or response lifecycle. Filters act as an
+extension point to express additional processing that may be performed in Gateway implementations. Some examples include
+request or response modification, implementing authentication strategies, rate-limiting, and traffic shaping.
+
+The following example adds header "my-header: foo" to HTTP requests with Host header "my.filter.com".
+```yaml
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: http-filter-1
+spec:
+  hostnames:
+    - my.filter.com
+  rules:
+    ...
+      filters:
+      - type: RequestHeaderModifier
+        requestHeaderModifier:
+          add:
+            my-header: foo
```

Comment:
Tangential, but seeing this example makes me wonder if we should have split this into 2 filter types - one to add headers and one to remove headers just for the sake of slightly more concise config.